lib/puppet/type/exec.rb

```diff
@@ -581,7 +581,16 @@ def check_all_attributes(refreshing = false)
           val = @parameters[check].value
           val = [val] unless val.is_a? Array
           val.each do |value|
-            return false unless @parameters[check].check(value)
+            if !@parameters[check].check(value)
+              # Give a debug message so users can figure out what command would have been
+              # but don't print sensitive commands or parameters in the clear
+              cmdstring = @parameters[:command].sensitive ? "[command redacted]" : @parameters[:command].value
+              checkstring = @parameters[check].sensitive ? "[redacted]" : @parameters[check].value
```

Comment:
Part of the sensitive confusion is some properties are defined to be sensitive in the type, like the [user's password](https://github.com/puppetlabs/puppet/blob/6.0.5/lib/puppet/type/user.rb#L258). Other times the manifest author defines a value as sensitive in their manifest, e.g. `command => Sensitive.new('secret command')`. To confuse matters https://github.com/puppetlabs/puppet/blob/6.0.5/lib/puppet/type.rb#L2413-L2440, says it works for properties, but not parameters. However `command` is a parameter, as are `onlyif` and `unless` are parameters. So we may just need to update https://github.com/puppetlabs/puppet/blob/6.0.5/lib/puppet/type/exec.rb#L618-L621 to flag `onlyif`/`unless` as sensitive?

Reply:
Yes, I was looking at the handling of command, and wondering if we need sensitive handling of the other checks for the exact same reasoning (some kind of secret, maybe a password, on the command line).